### PR TITLE
Added more crumb logging

### PR DESCRIPTION
**Pull request by iprak**
Added more logging around crumb requests. The crumb response content will also be saved to YahooFinanceCrumbContent.log for failed case when Info level logging is enabled.

### CODE REVIEW

**Comment by iprak** [comment]
Hello @swiergot , would you mind running this at your end. This pull just includes more logging and unit tests.

File: custom_components/yahoofinance/coordinator.py [hunk 1]
```diff
@@ -257,67 +277,82 @@ def add_symbol(self, symbol: str) -> bool:
...
+        url = await self.build_request_url()
+        cookies = self._cc.cookies
         _LOGGER.debug("Requesting data from '%s'", url)
 
         try:
             async with async_timeout.timeout(WEBSESSION_TIMEOUT):
-                headers = {"User-Agent": USER_AGENT}
                 response = await self.websession.get(
-                    url, headers=headers, cookies=cookies
+                    url, headers=REQUEST_HEADERS, cookies=cookies
```
Comment (swiergot) [inline custom_components/yahoofinance/coordinator.py:287]: If crumb contains a slash, will it be automatically encoded? I had urllib.parse.quote for that which you have removed.

**Comment by swiergot** [comment]
Works fine, also when crumb has a slash.

**Comment by iprak** [comment]
> Works fine, also when crumb has a slash.

Thanks

**Pull request merged**